packages/virtual-core/src/index.ts

```diff
@@ -464,16 +470,68 @@ export class Virtualizer<
 
       for (let i = min; i < count; i++) {
         const key = getItemKey(i)
-        const measuredSize = itemSizeCache.get(key)
-        const start = measurements[i - 1]
-          ? measurements[i - 1]!.end
+
+        const complete = new Map<number, true>()
+        const columns = new Map<number, VirtualItem>()
+        for (let m = i - 1; m >= 0; m--) {
+          if (complete.size === this.options.columns) {
+            break
+          }
+
+          const measurement = measurements[m]!
+          const index = this.options.horizontal
+            ? measurement.top
+            : measurement.left
+
+          if (complete.has(index)) {
+            continue
+          }
+
+          const previous = columns.get(index)
+          if (previous == null || measurement.end > previous.end) {
+            columns.set(index, measurement)
+          } else if (measurement.end < previous.end) {
+            complete.set(index, true)
+          }
+        }
```

Comment:
> Wondering how big impact on getMeasurements performance would this calculations do.

@piecyk I did a few things to try to minimize the performance impact on the calculations. The first thing is that it loops over the elements starting from the end since the element we are looking for is very likely to be among the lasts. The second thing is that once it found a column's last element, it ignores other elements from that column. Finally, it stops looping over the elements when every columns' last element has been found.

For lists not using masonry (so only 1 column), it will always loop only over the last element so it should be almost as fast as before (I think).

With that being said I haven't run benchmarks and I may be missing the implication of this change on other parts of the code.